apps/algolia/lib/algolia/update.ex

```diff
@@ -13,14 +13,43 @@ defmodule Algolia.Update do
   """
   @spec update(String.t() | nil) :: t
   def update(host \\ nil) do
-    Map.new(@indexes, &{&1, update_index(&1, host)})
+    Map.new(@indexes, &{&1, update_and_clean_index(&1, host)})
   end
 
-  @spec update_index(atom, String.t()) :: success | error
-  def update_index(index_module, base_url) do
-    index_module.all()
-    |> Enum.filter(&has_routes?/1)
-    |> Enum.map(&build_data_object/1)
+  @spec update_and_clean_index(atom, String.t()) :: :ok | error
+  def update_and_clean_index(index_module, base_url) do
+    new_objects =
+      index_module.all()
+      |> Enum.filter(&has_routes?/1)
+      |> Enum.map(&to_data_object/1)
+
+    with {:ok, _body} <- clean_index(new_objects, index_module, base_url),
+         {:ok, _body} <- update_index(new_objects, index_module, base_url) do
+      :ok
+    end
+  end
+
+  @spec clean_index(Enum.t(), atom, String.t()) :: success | error
+  def clean_index(new_objects, index_module, base_url) do
+    current_objs = current_objects(base_url, index_module)
```

Comment:
Small thing, but above we call them new_objects and here current_objs. I'm guessing that is because current_objects is a function. But, it indicates we need some better naming maybe:

```
current_objects = get_current_objects(base_url, index_module)
```

But, I see there is also a get_current_objects. So, maybe combine what the two are doing. Currently, it's hard to read.

Reply:
I will work on renaming.  `get_current_objects` is the name for the function that gets all the objects from algiolas endpoint